<?xml version="1.0" encoding="UTF-8"?>
<root BTCPP_format="4">
  <BehaviorTree ID="Untitled">
    <ReactiveSequence>
      <SubGameStatus topic_name="game_status"
                     game_status="{game_status}"/>
      <WhileDoElse>
        <IsGameTime message="{game_status}"
                    game_progress="4"/>
        <SendGoal goal_pose="0;0;0; 0;0;0;1"
                  action_name="navigate_to_pose"/>
        <SendGoal goal_pose="5;-3.5;0; 0;0;0;1"
                  action_name="navigate_to_pose"/>
      </WhileDoElse>
    </ReactiveSequence>
  </BehaviorTree>

  <!-- Description of Node Models (used by Groot) -->
  <TreeNodesModel>
    <Action ID="IsGameTime"
            editable="true">
      <input_port name="message"
                  default="{game_status}"/>
      <input_port name="game_progress"
                  default="4"/>
    </Action>
    <Action ID="SendGoal"
            editable="true">
      <input_port name="goal_pose"
                  default="0;0;0; 0;0;0;1"/>
      <input_port name="action_name"
                  default="navigate_to_pose"/>
    </Action>
    <Action ID="SubGameStatus"
            editable="true">
      <input_port name="topic_name"
                  default="game_status"/>
      <output_port name="game_status"
                   default="{game_status}"/>
    </Action>
  </TreeNodesModel>

</root>